### PR TITLE
ISPN-12828 HotRod client receives a server topology before the server…

### PR DESCRIPTION
**Pull request by jabolina**
https://issues.redhat.com/browse/ISPN-12828

… started

### CODE REVIEW

**Comment by danfaer** [comment]
Looks good, thank you! As a part of the fix for the same problem WDYT about adding a configurable retry for 

https://github.com/infinispan/infinispan/blob/cea9ef6ac5b50c9a8d5f3f95a3e9a9a432f5ffcf/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelFactory.java#L522-L522

**Comment by wburns** [comment]
> Looks good, thank you! As a part of the fix for the same problem WDYT about adding a configurable retry for
> 
> https://github.com/infinispan/infinispan/blob/cea9ef6ac5b50c9a8d5f3f95a3e9a9a432f5ffcf/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelFactory.java#L522-L522

@danfaer 
I don't think that is worth it. This code here is just trying to preemptively open the connection to any newly added servers, so that a user call doesn't have to be the first to do so.

File: server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java [hunk 1]
```diff
@@ -123,6 +125,16 @@ public void start(C configuration, EmbeddedCacheManager cacheManager) {
       }
    }
 
+   @Override
+   public final void postStart() {
+      try {
+         internalPostStart();
+      } catch (RuntimeException e) {
```
Comment (wburns) [inline server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java:132]: Should this be Throwable ?

**Review by wburns — APPROVED** [review]
LGTM.

File: server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java (hunk 1 above)
Comment (jabolina) [reply server/core/src/main/java/org/infinispan/server/core/AbstractProtocolServer.java:132]: This was mimicking the `start` method above. I can change it, but we'll need to wrap it on a runtime exception.

**Review by wburns — APPROVED** [review]

**Comment by wburns** [comment]
Integrated into main, thanks @jabolina !

**Comment by danfaer** [comment]
Thanks!

**Comment by danfaer** [comment]
@wburns Can this PR be cherrypicked into 15.0.x ?

**Comment by wburns** [comment]
> @wburns Can this PR be cherrypicked into 15.0.x ?

I created https://github.com/infinispan/infinispan/pull/13132, but it might not get in for the next build.

**Pull request merged**